src/reader.rs

```diff
@@ -117,34 +118,35 @@ impl TIFFReader {
-    /// This means that the tag values have to be read taking the tag type into consideration.
-    fn vec_to_tag_value<Endian: ByteOrder>(&self, vec: Vec<u8>, tpe: &Type) -> TagValue {
+    /// Converts a Vec<u8> into a tiff::decoder::ifd::Value, depending on the type of the tag. In
+    /// the TIFF file format, each tag type indicates which value it stores (e.g., a byte, ascii,
+    /// or long value). This means that the tag values have to be read taking the tag type into
+    /// consideration.
+    fn vec_to_tag_value<Endian: ByteOrder>(&self, vec: Vec<u8>, tpe: &Type) -> Value {
         let len = vec.len();
         match tpe {
-            Type::BYTE => TagValue::ByteValue(vec[0]),
-            Type::ASCII => TagValue::AsciiValue(String::from_utf8_lossy(&vec).to_string()),
-            Type::SHORT => TagValue::ShortValue(Endian::read_u16(&vec[..])),
-            Type::LONG => TagValue::LongValue(Endian::read_u32(&vec[..])),
-            Type::RATIONAL => TagValue::RationalValue((
+            Type::BYTE => Value::Byte(vec[0]),
+            Type::ASCII => Value::Ascii(String::from_utf8_lossy(&vec).to_string()),
+            Type::SHORT => Value::Short(Endian::read_u16(&vec[..])),
+            Type::LONG => Value::Unsigned(Endian::read_u32(&vec[..])),
+            Type::RATIONAL => Value::Rational(
                 Endian::read_u32(&vec[..(len / 2)]),
                 Endian::read_u32(&vec[(len / 2)..]),
-            )),
-            Type::SBYTE => TagValue::SignedByteValue(vec[0] as i8),
-            Type::UNDEFINED => TagValue::ByteValue(0),
-            Type::SSHORT => TagValue::SignedShortValue(Endian::read_i16(&vec[..])),
-            Type::SLONG => TagValue::SignedLongValue(Endian::read_i32(&vec[..])),
-            Type::SRATIONAL => TagValue::SignedRationalValue((
+            ),
+            // Type::SBYTE => Value::SignedByte(vec[0] as i8), // TODO wait for upstream SignedByte enum variant
+            Type::UNDEFINED => Value::Byte(0),
+            // Type::SSHORT => Value::SignedShort(Endian::read_i16(&vec[..])), // TODO wait for upstream SignedShort enum variant
+            Type::SLONG => Value::Signed(Endian::read_i32(&vec[..])),
+            Type::SRATIONAL => Value::SRational(
                 Endian::read_i32(&vec[..(len / 2)]),
                 Endian::read_i32(&vec[(len / 2)..]),
-            )),
-            Type::FLOAT => TagValue::FloatValue(Endian::read_f32(&vec[..])),
-            Type::DOUBLE => TagValue::DoubleValue(Endian::read_f64(&vec[..])),
-            Type::IFD => unimplemented!(),
+            ),
+            Type::FLOAT => Value::Float(Endian::read_f32(&vec[..])),
+            Type::DOUBLE => Value::Double(Endian::read_f64(&vec[..])),
+            Type::IFD => Value::Ifd(Endian::read_u32(&vec[..])),
             Type::LONG8 => unimplemented!(),
             Type::SLONG8 => unimplemented!(),
-            Type::IFD8 => unimplemented!(),
+            Type::IFD8 => Value::IfdBig(Endian::read_u64(&vec[..])),
```

Comment:
Double check if `read_u32` and `read_u64` here is ok. Based on https://github.com/image-rs/image-tiff/blob/0c54a18e2130bd8e3e897009e1fb59eaaf607c6c/src/decoder/ifd.rs#L438-L442.